### PR TITLE
Reducing the number of variants for ray tracing when building a player

### PR DESCRIPTION
**Pull request by anisunity**
Replaced the DIFFUSE_LIGHTING_ONLY multicompile by a uniform.
Removed the dynamic lightmap multicompile (was un-used, but still induced a lot of shader compilation)
Removed the LOD cross fade multi compile for ray tracing (was un-used, but still induced a lot of shader compilation)
Added a ray tracing mode option in the HDRP asset that allows to override and shader stripping.
When in something else than both, volume components only display the current configuration parameters for indirect diffuse and reflection.

This was done to reduce the build time from scratch for the player (example being smallofficeraytracing).

DXR tests are still green locally.

https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/6432

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/ShaderGraph/HDKeywords.cs [hunk 1]
```diff
@@ -79,15 +79,6 @@ public static class Descriptors
                 scope = KeywordScope.Global,
             };
 
-            public static KeywordDescriptor DiffuseLightingOnly = new KeywordDescriptor()
```
Comment (anisunity) [inline com.unity.render-pipelines.high-definition/Editor/ShaderGraph/HDKeywords.cs:82]: @alelievr change to SG here

Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Editor/ShaderGraph/HDKeywords.cs:82]: Thanks !

File: com.unity.render-pipelines.high-definition/Editor/ShaderGraph/HDTarget.cs [hunk 2]
```diff
@@ -541,6 +541,15 @@ static class CoreKeywords
             { CoreKeywordDescriptors.AlphaTest, new FieldCondition(Fields.AlphaTest, true) },
         };
 
+        public static KeywordCollection HDBaseNoCrossFade = new KeywordCollection
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/ShaderGraph/HDTarget.cs:544]: poking @Kink3d and @alelievr about this PR that affect shader graph and have been merged into HDRP/staging

**Comment by sebastienlagarde** [comment]
poking @Kink3d and @alelievr about this PR that affect shader graph and have been merged into HDRP/staging

**Pull request merged**